module/VuFindSearch/src/VuFindSearch/Backend/Summon/GuzzleConnector.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * Summon Search API Interface (Guzzle and Psr implementation)
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFindSearch\Backend\Summon;
+
+use GuzzleHttp\Client as HttpClient;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use SerialsSolutions_Summon_Exception;
+
+use function is_object;
+
+/**
+ * Guzzle and PSR-compliant port of SerialsSolutions\Summon\Laminas connector
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Sambhav Pokharel <sambhavpokharel@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class GuzzleConnector extends \SerialsSolutions_Summon_Base implements LoggerAwareInterface
+{
+    /**
+     * HTTP client instance
+     *
+     * @var HttpClient
+     */
+    protected $client;
+
+    /**
+     * Logger instance.
+     *
+     * @var ?LoggerInterface
+     */
+    protected $logger = null;
+
+    /**
+     * Constructor.
+     *
+     * @param string     $apiId   Summon API ID
+     * @param string     $apiKey  Summon API Key
+     * @param array      $options Options for the parent constructor
+     * @param HttpClient $client  Optional HTTP client to use
+     */
+    public function __construct(string $apiId, string $apiKey, array $options = [], ?HttpClient $client = null)
+    {
+        parent::__construct($apiId, $apiKey, $options);
+        $this->client = $client ?? new HttpClient();
+    }
+
+    /**
+     * Sets the logger instance.
+     *
+     * @param LoggerInterface $logger The logger instance to set.
+     *
+     * @return void
+     */
+    public function setLogger(LoggerInterface $logger): void
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Prints a debug message if debug is enabled.
+     *
+     * @param string $msg The message to debug.
+     *
+     * @return void
+     */
+    protected function debugPrint($msg)
+    {
+        if ($this->logger) {
+            $this->logger->debug("$msg");
```

Comment:
Quotes are harmless but unnecessary here:

```suggestion
            $this->logger->debug($msg);
```